suite.go

```diff
@@ -109,87 +106,85 @@ func (s *suite) runStep(ctx context.Context, pickle *Scenario, step *Step, prevS
 
 		switch err {
 		case nil:
-			sr.Status = models.Passed
-			s.storage.MustInsertPickleStepResult(sr)
+			stepResult.Status = models.Passed
+			s.storage.MustInsertPickleStepResult(stepResult)
 
-			s.fmt.Passed(pickle, step, match.GetInternalStepDefinition())
+			s.fmt.Passed(pickle, step, stepDefinition.GetInternalStepDefinition())
 		case ErrPending:
-			sr.Status = models.Pending
-			s.storage.MustInsertPickleStepResult(sr)
+			stepResult.Status = models.Pending
+			s.storage.MustInsertPickleStepResult(stepResult)
 
-			s.fmt.Pending(pickle, step, match.GetInternalStepDefinition())
+			s.fmt.Pending(pickle, step, stepDefinition.GetInternalStepDefinition())
 		default:
-			sr.Status = models.Failed
-			sr.Err = err
-			s.storage.MustInsertPickleStepResult(sr)
+			stepResult.Status = models.Failed
+			stepResult.Err = err
+			s.storage.MustInsertPickleStepResult(stepResult)
 
-			s.fmt.Failed(pickle, step, match.GetInternalStepDefinition(), err)
+			s.fmt.Failed(pickle, step, stepDefinition.GetInternalStepDefinition(), err)
 		}
 	}()
 
-	// run before scenario handlers
 	if isFirst {
 		ctx, err = s.runBeforeScenarioHooks(ctx, pickle)
 	}
 
-	// run before step handlers
 	ctx, err = s.runBeforeStepHooks(ctx, step, err)
 
-	match = s.matchStep(step)
-	s.storage.MustInsertStepDefintionMatch(step.AstNodeIds[0], match)
-	s.fmt.Defined(pickle, step, match.GetInternalStepDefinition())
+	stepDefinition = s.matchStep(step)
+	s.storage.MustInsertStepDefintionMatch(step.AstNodeIds[0], stepDefinition)
+	s.fmt.Defined(pickle, step, stepDefinition.GetInternalStepDefinition())
 
 	if err != nil {
-		sr = models.NewStepResult(pickle.Id, step.Id, match)
-		sr.Status = models.Failed
-		s.storage.MustInsertPickleStepResult(sr)
+		stepResult = models.NewStepResult(pickle.Id, step.Id, stepDefinition)
+		stepResult.Status = models.Failed
+		s.storage.MustInsertPickleStepResult(stepResult)
 
 		return ctx, err
 	}
 
 	if ctx, undef, err := s.maybeUndefined(ctx, step.Text, step.Argument); err != nil {
 		return ctx, err
 	} else if len(undef) > 0 {
-		if match != nil {
```

Comment:
I think this rename takes away some of the context.

The fact that this var is a step definition can be derived from its type, so the name does not convey additional information. However this var holds a specific step definition: a particular match.

I think it would be better to keep original name to reflect the purpose rather than content.

Reply:
True. And would `matchingStepDefinition` be too long-winded a name?

Reply:
It is fine for me, though in general shorter names are preferred as long as they tell enough and don't bring confusion (e.g. having two different matches of different kinds in the same scope would require more specific names). 

https://research.swtch.com/names

Reply:
Out of interest, what does that `undef` variable there above contain? Is that the list of undefined, _nested_, steps?

Reply:
I checked using tests, it holds a list of one item, a string with step text, e.g. `another undefined step`.